tests/manage/monitoring/conftest.py

```diff
@@ -64,51 +73,98 @@ def prometheus_log(info, alert_list):
             time.sleep(3)
         logger.info('Logging of all prometheus alerts stopped')
 
-    if not measure_after:
-        start_time = time.time()
+    if os.path.isfile(result_file) and os.access(result_file, os.R_OK):
+        logger.info(
+            f"File {result_file} already created."
+            f" Trying to use it for tests..."
+        )
+        with open(result_file) as open_file:
+            results = json.load(open_file)
+        logger.info(
+            f"File {result_file} loaded. Content of file:\n{results}"
+        )
 
-    # init logging thread that checks for Prometheus alerts
-    # while workload is running
-    # based on https://docs.python.org/3/howto/logging-cookbook.html#logging-from-multiple-threads
-    info = {'run': True}
-    alert_list = []
+    else:
+        logger.info(
+            f"File {result_file} not created yet. Starting measurement..."
+        )
+        if not measure_after:
+            start_time = time.time()
 
-    logging_thread = threading.Thread(
-        target=prometheus_log,
-        args=(info, alert_list)
-    )
-    logging_thread.start()
-
-    result = operation()
-    if measure_after:
-        start_time = time.time()
-    passed_time = time.time() - start_time
-    if minimal_time:
-        additional_time = minimal_time - passed_time
-        if additional_time > 0:
-            time.sleep(additional_time)
-    stop_time = time.time()
-    info['run'] = False
-    logging_thread.join()
-    logger.info(f"Alerts found during measurement: {alert_list}")
-    return {
-        'start': start_time,
-        'stop': stop_time,
-        'result': result,
-        'metadata': metadata,
-        'prometheus_alerts': alert_list
-    }
-
-
-@pytest.fixture(scope="session")
-def workload_stop_ceph_mgr():
+        # init logging thread that checks for Prometheus alerts
+        # while workload is running
+        # based on https://docs.python.org/3/howto/logging-cookbook.html#logging-from-multiple-threads
+        info = {'run': True}
+        alert_list = []
+
```

Comment:
Please add comments in the code in this function

Reply:
I added some comments. Please let me know if there needs to be added more comments to more parts of code.